source/simulator/assemblers/newton_stokes.cc

```diff
@@ -414,8 +414,8 @@ namespace aspect
                         for (unsigned int j=0; j<stokes_dofs_per_cell; ++j)
                           {
                             data.local_matrix(i,j) += ( derivative_scaling_factor * alpha * (scratch.grads_phi_u[i] * deta_deps_times_eps_times_phi[j]
-                                                                                             + scratch.grads_phi_u[j] * deta_deps_times_eps_times_phi[i])
-                                                        + derivative_scaling_factor * pressure_scaling * 2.0 * viscosity_derivative_wrt_pressure * scratch.phi_p[j] * scratch.grads_phi_u[i] * strain_rate )
+                                                                                             + (scratch.grads_phi_u[j] * deta_deps_times_eps_times_phi[i]))
+                                                        + derivative_scaling_factor * pressure_scaling * 2.0 * viscosity_derivative_wrt_pressure * scratch.phi_p[j] * (scratch.grads_phi_u[i] * strain_rate) )
```

Comment:
This, too, should make no difference because multiplication is distributive. But, it's quite possible that doing the dot product between matrices first,  and then multiplying with scalars (in the second of these lines) is substantially faster than multiplying a tensor repeatedly by scalars, and only then doing the dot product.

Reply:
... but this can lead to floating point differences.